src/main/java/com/bootcamp/demo/restapi/UserController.java

```diff
@@ -0,0 +1,78 @@
+package com.bootcamp.demo.restapi;
+
+import com.bootcamp.demo.model.User;
+import com.bootcamp.demo.service.UserService;
+import com.bootcamp.demo.service.exception.ItemNotFoundException;
+import com.bootcamp.demo.service.exception.ServiceException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Set;
+
+
+@RestController
+@RequestMapping(path = "/api/users")
+public class UserController {
+
+    private final UserService userService;
+
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @PostMapping
+    public ResponseEntity<Object> postUser(@RequestBody final User user) {
+        try {
+            userService.createUser(user);
+            return new ResponseEntity<>(HttpStatus.CREATED);
+        } catch (ServiceException e) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @DeleteMapping("/{id}")
+    @ResponseBody
+    public ResponseEntity<Object> deleteUserById(@PathVariable(value = "id") final String id) {
+        try {
+            userService.deleteUserById(id);
+            return new ResponseEntity<>(HttpStatus.ACCEPTED);
+        } catch (ServiceException e) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @GetMapping("/getUserById/{id}")
+    @ResponseBody
+    public ResponseEntity<Object> getUserById(@PathVariable(value = "id") final String id) {
+        try {
+            return new ResponseEntity<>(userService.getUserById(id), HttpStatus.OK);
+        } catch (ItemNotFoundException e) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        } catch (ServiceException e) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @GetMapping("/getUserByEmail/{email}")
+    @ResponseBody
+    public ResponseEntity<Object> getUserByEmail(@PathVariable(value = "email") final String email) {
```

Comment:
Notice the duplication between this method and `getUserById`. In order to avoid it, you could extract this to another method which could look like the below:
```
private ResponseEntity<Object> getBy(final Supplier<User> supplier) { // with better/more intuitive name for the method and the param
    try {
        return new ResponseEntity<>(supplier.get(), HttpStatus.OK);
    } catch (ItemNotFoundException e) {
        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
    } catch (ServiceException e) {
        return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
    }
}
```
and would be called as `return getBy(() -> userService.getUserById(id));` from `getUserById` and similarly from `getUserByEmail`.